gla_cta.module

```diff
@@ -11,21 +11,53 @@ use Drupal\Core\Form\FormStateInterface;
  * Implements hook_field_widget_WIDGET_TYPE_form_alter().
  */
 function gla_cta_field_widget_paragraphs_form_alter(&$element, FormStateInterface $form_state, $context) {
+  /** @var \Drupal\gla_core_profile\FormHelper $formHelper */
+  $formHelper = \Drupal::service('gla_core_profile.form_helper');
 
-  // Use Drupal States to hide the image field if the CTA type is not image.
   if ($element['#paragraph_type'] == 'cta') {
-    if (isset($element['subform']['field_p_c_image'])) {
-      if (!empty($element['subform']['#parents'])) {
-        $parents = $element['subform']['#parents'];
-        $name = array_shift($parents) . '[';
-        $name .= implode('][', $parents);
-        $name .= '][field_p_c_type]';
-        $element['subform']['field_p_c_image']['#states'] = [
+    if (!empty($element['subform']['#parents'])) {
+      $cta_type_name = $formHelper->buildNameSelector($element['subform']['field_p_cta_type']);
+
+      // Only show the image field for image CTA types.
+      if (isset($element['subform']['field_p_cta_image'])) {
+        $element['subform']['field_p_cta_image']['#states'] = [
           'visible' => [
-            ':input[name="' . $name . '"]' => ['value' => 'image'],
+            ':input[name="' . $cta_type_name . '"]' => ['value' => 'image'],
           ],
         ];
       }
+
+      // Only show the summary field for image CTA types.
+      if (isset($element['subform']['field_p_cta_summary'])) {
+        $element['subform']['field_p_cta_summary']['#states'] = [
+          'visible' => [
+            ':input[name="' . $cta_type_name . '"]' => ['value' => 'image'],
+          ],
+        ];
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_paragraph().
+ */
+function gla_cta_preprocess_paragraph(&$variables) {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  if ($paragraph->bundle() == 'cta') {
+    $cta_type = $paragraph->field_p_cta_type->first()->value;
```

Comment:
Does this code fail if the first() value is not set? Or if there is no value set?

I would add a check to exit out of the code if the first() value doesn't exist.

Something like this on line 48 might fail gracefully.

if (!isset($paragraph->field_p_cta_type->first())) {
  return;
}

Reply:
The field is required so technically there's always a value

Reply:
It is good to have defensive code.
That required can change in the future and not have context of this dependency.

Reply:
I think this is probably fine as it is, but I've changed it to `$cta_type = $paragraph->field_p_cta_type->value;` as the `first()` isn't necessary, Drupal will give us the first value by default, and it's not a multi-value field anyway.

If the `field_p_cta_type` field is made optional in the future and we are processing a Paragraph with no `field_p_cta_type` value then the following if statement would be skipped  anyway, so I don't think there's any need to return early - the if statement is preventing the `field_p_cta_type = image` specific code from being executed for us already.